ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -195,6 +201,18 @@ private PeerDiscoveryController(
             "type");
   }
 
+  private void onPeerAdded(final Peer peer, final boolean wasAdded) {
+    if (wasAdded) {
+      addToPeerTable(DiscoveryPeer.fromEnode(peer.getEnodeURL()));
+    }
+  }
+
+  private void onPeerRemoved(final Peer peer, final boolean wasRemoved) {
+    if (wasRemoved) {
+      peerTable.tryEvict(peer);
```

Comment:
Do we need this logic?  Honestly not sure if we do or not, but it's little unexpected at first glance.